### PR TITLE
EZP-32239: Data providers and eContent revision

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/EZP-32239
| Versions      | 3.x, master

Note: "Content model data provider" (ez5 data provider in code) renamed to "Repository data provider" after consultation with @alongosz 

To do: 
- [ ] Verify the status of Catalog Segmentation feature

### CODE REVIEW

File: docs/guide/econtent/econtent.md [hunk 1]
```diff
@@ -1,39 +1,19 @@
...
-[[= product_name_com =]] uses an almost generic way to access the catalog.
-The catalog can be stored in the content model or in eContent. 
+## eContent Back Office [[% include 'snippets/commerce_badge.md' %]]
 
-If eContent is used, there are a few restrictions that have to be considered:
+You can see an overview of the data model for eContent in the Back Office in **Control center**.
 
-- A product cannot be embedded by using the standard embed feature. [[= product_name_com =]] offers an alternative feature which enables embedding products in RichtText Fields.
-- eContent products are not visible in the Back Office.
+The model itself has to be defined using SQL. The tab **eContent types** shows an overview of:
```
Comment (dabrt) [inline docs/guide/econtent/econtent.md:16]:
```suggestion
The model itself has to be defined by using SQL. 
The **eContent types** tab shows an overview of:
```

File: docs/guide/econtent/econtent.md [hunk 2]
```diff
@@ -1,39 +1,19 @@
...
-| Flexible data model                   | yes  | yes  |
-| Translations                         | yes  | yes  |
-| Interface for editing                | yes  | no |
-| Versioning                           | yes  | no |
-| Simple segmentation          | yes  | yes  |
-| Extended segmentation        | no | yes  |
-| Fast imports (e.g. from PIM)         | no | yes  |
-| Support of large catalogs              | no | yes  |
-| Staging (live and temporary space) | no | yes  |
+- defined eContent types (e.g. `product_group`, `product`)
```
Comment (dabrt) [inline docs/guide/econtent/econtent.md:18]:
```suggestion
- the defined eContent types (for example, `product_group` or `product`)
```

File: docs/guide/econtent/indexing_econtent_data.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+eContent Solr configuration uses two cores to keep search services available while indexing.
+This means that the indexer indexes eContent data in the back core,
+and then the indexer can be executed again with a swap parameter to swap the cores.
+
+|Command options|Results|
+|--- |--- |
+|`-c 5000`|Specifies the chunk size (how many elements are indexed at the same time).|
+|`-r`|Deletes all previous eContent data.|
+|`--siteaccess=import`|Informs the indexer to take the product data from the temporary tables.|
```
Comment (dabrt) [inline docs/guide/econtent/indexing_econtent_data.md:17]:
```suggestion
|`--siteaccess=import`|Instructs the indexer to take the product data from the temporary tables.|
```

File: docs/guide/econtent/indexing_econtent_data.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+|`--live-core`|Indexes directly on the live core (and not on the back core).</br>If you use `-r` together `--live-core`, the delete operation takes place before starting the indexer, so the full indexed content is not be available while indexing.|
```
Comment (dabrt) [inline docs/guide/econtent/indexing_econtent_data.md:18]:
```suggestion
|`--live-core`|Indexes directly on the live core (and not on the back core).</br>If you use `-r` together `--live-core`, the delete operation takes place before the indexer starts, so the full indexed content is not be available while indexing.|
```

File: docs/guide/econtent/indexing_econtent_data.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+|`swap`|Does not index, but swaps the cores instead.|
+
+### Additional parameters
+
+|||
+|--- |--- |
+|`siso_search.default.index_econtent_languages`|Specifies the languages to index.|
+|`siso_search.default.solr_spellcheck`|Enables or disables Solr spellcheck (aka "Did you mean?" functionality).|
```
Comment (dabrt) [inline docs/guide/econtent/indexing_econtent_data.md:26]:
```suggestion
|`siso_search.default.solr_spellcheck`|Enables or disables the Solr spellchecker (aka the "Did you mean?" functionality).|
```

File: docs/guide/econtent/staging_system.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# Staging system
+
+eContent supports a staging area. The product catalog can exist either as a live version which is used for the customer,
```
Comment (dabrt) [inline docs/guide/econtent/staging_system.md:3]:
```suggestion
eContent supports a staging area. 
The product catalog can exist either as a live version which is used for the customer,
```

File: docs/guide/econtent/staging_system.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+Use a SiteAccess (default: `import`) for the indexing process.
+The import SiteAccess should cover all languages used in eContent (see `ezplatform.yaml`).
+By default, the table set for this SiteAccess is configured to use the temporary tables
+
+The import process does not stop the production system.
+
+## Switching between temporary and live versions
+
+After the import, switch the database tables and Solr cores. Execute the following commands:
```
Comment (dabrt) [inline docs/guide/econtent/staging_system.md:23]:
```suggestion
After the import, switch the database tables and Solr cores. 
Execute the following commands:
```

File: docs/guide/econtent/staging_system.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+``` bash
+php bin/console ibexa:commerce:index-econtent swap
+php bin/console ibexa:commerce:swap-econtent-tables
+```
+
+Depending on the project, you should purge the HTTP cache for the product catalog.
```
Comment (dabrt) [inline docs/guide/econtent/staging_system.md:30]: "Depending on the project" suggest choice while "You should" suggests neccessity. 

How about 
"Depending on the project, you might want to purge the HTTP cache for the product catalog."

When would I want to do this? What kind of project might require doing so?

File: docs/guide/shop_features/catalog.md [hunk 9]
```diff
@@ -8,11 +8,11 @@ The source of products depends on the data provider you have configured.
 
 Data providers define where product information is stored.
 
-You can use one of two data providers: the content model data provider and the eContent data provider.
+You can use one of two data providers: the Repository data provider and the eContent data provider.
```
Comment (dabrt) [inline docs/guide/shop_features/catalog.md:11]:
```suggestion
You can use one of the following data providers: the Repository data provider and the eContent data provider.
```

**Review by dabrt — CHANGES_REQUESTED** [review]

File: docs/guide/econtent/staging_system.md (hunk 8 above)
Comment (DominikaK) [inline docs/guide/econtent/staging_system.md:30]:
```suggestion
Depending on the project, you might want to purge the HTTP cache for the product catalog.
```

File: docs/guide/econtent/indexing_econtent_data.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+|`--siteaccess=import`|Instructs the indexer to take the product data from the temporary tables.|
+|`--live-core`|Indexes directly on the live core (and not on the back core).</br>If you use `-r` together `--live-core`, the delete operation takes place before the indexer starts, so the full indexed content is not be available while indexing.|
+|`swap`|Does not index, but swaps the cores instead.|
+
+### Additional parameters
+
+|||
+|--- |--- |
+|`siso_search.default.index_econtent_languages`|Specifies the languages to index.|
+|`siso_search.default.solr_spellcheck`|Enables or disables the Solr spellchecker (aka the "Did you mean?" functionality).|
```
Comment (dabrt) [inline docs/guide/econtent/indexing_econtent_data.md:26]: I am thinking whether the "aka" is not overly colloqial. I could be used in legal documents or criminal records but in tech doc I think "i.e." could fit better. WDYT?

**Review by dabrt — APPROVED** [review]
Please see one comment

File: docs/guide/econtent/indexing_econtent_data.md (hunk 10 above)
Comment (DominikaK) [inline docs/guide/econtent/indexing_econtent_data.md:26]:
```suggestion
|`siso_search.default.solr_spellcheck`|Enables or disables the Solr spellchecker (the "Did you mean?" functionality).|
```

**Pull request merged**